include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td

```diff
@@ -471,6 +471,46 @@ def NVMMASharedEncodingAttr :
   let hasCustomAssemblyFormat = 1;
 }
 
+def SwizzledBlocksSharedEncodingAttr :
+  TritonGPU_Attr<"SwizzledBlocksSharedEncoding", "swizzled_blocks_shared_encoding", [SharedEncodingTrait, LayoutEncodingTrait]> {
+  let mnemonic = "swizzled_blocks_shared";
+
+  let description = [{
+Shared encoding similar to SwizzledSharedEncodingAttr,
+but with additional swizzling pattern change in blocks of maxPhase * perPhase rows:
+
+  #shared<{vec=1, perPhase=1, maxPhase=4, order=[1, 0]}>
+  [ 0,  1,  2,  3],  // xor with 0  (phase = 0 ^ block no = 0)
+  [ 5,  4,  7,  6],  // xor with 1  (phase = 1 ^ block no = 0)
+  [10, 11,  8,  9],  // xor with 2  (phase = 2 ^ block no = 0)
+  [15, 14, 13, 12]   // xor with 3  (phase = 3 ^ block no = 0)
+  [17, 16, 19, 18],  // xor with 1  (phase = 0 ^ block no = 1)
+  [20, 21, 22, 23],  // xor with 0  (phase = 1 ^ block no = 1)
+  [27, 26, 25, 24],  // xor with 3  (phase = 2 ^ block no = 1)
+  [30, 31, 28, 29]   // xor with 2  (phase = 3 ^ block no = 1)
+
+Used to minimize bank conflicts while loading MFMA operand B in TN GEMM case
```

Comment:
Not sure this comment is correct? I think we need this for non-TN case? Anyway, let's not use the term "TN" here given it can be confusing given different conext means different interpretations. Let's just explain that this is in general for load from global memory with a different major-ness than what we write to shared memory with transposes in threads' registers when no direct hardware transposed shared memory load/store.